### PR TITLE
CLI-ify and improve modelFor docs

### PR DESCRIPTION
**Pull request by alexspeller**
It's generally not a good idea to make stateful changes in validation hooks as
the transition can still get rolled back - this fixes that and also cli-ifys
that part of the doc

### CODE REVIEW

**Review by locks — CHANGES_REQUESTED** [review]
Can you do the whole file? 👼

File: packages/ember-routing/lib/system/route.js [hunk 1]
```diff
@@ -1770,16 +1770,20 @@ let Route = EmberObject.extend(ActionHandler, Evented, {
...
 
     ```javascript
-    App.Router.map(function() {
-        this.route('post', { path: '/post/:post_id' }, function() {
-          this.route('comments', { resetNamespace: true });
-        });
+    // app/router.js
+    Router.map(function() {
+      this.route('post', { path: '/post/:post_id' }, function() {
+        this.route('comments', { resetNamespace: true });
```
Comment (locks) [inline packages/ember-routing/lib/system/route.js]: 👎  `resetNamespace`

File: packages/ember-routing/lib/system/route.js [hunk 2]
```diff
@@ -1770,16 +1770,20 @@ let Route = EmberObject.extend(ActionHandler, Evented, {
...
     });
 
-    App.CommentsRoute = Ember.Route.extend({
-        afterModel: function() {
-          this.set('post', this.modelFor('post'));
-        }
+
+    // app/routes/comments.js
+    export default Ember.Route.extend({
+      afterModel: function() {
```
Comment (locks) [inline packages/ember-routing/lib/system/route.js]: ES2015 `afterModel() {`

File: packages/ember-routing/lib/system/route.js [hunk 3]
```diff
@@ -1770,16 +1770,20 @@ let Route = EmberObject.extend(ActionHandler, Evented, {
...
+      this.route('post', { path: '/post/:post_id' }, function() {
+        this.route('comments', { resetNamespace: true });
+      });
     });
 
-    App.CommentsRoute = Ember.Route.extend({
-        afterModel: function() {
-          this.set('post', this.modelFor('post'));
-        }
+
```
Comment (locks) [inline packages/ember-routing/lib/system/route.js]: I'd maybe have two different snippets to make it more evident they're different files.

**Review by locks — CHANGES_REQUESTED** [review]

**Comment by alexspeller** [comment]
@locks updated the whole file now 👍

**Review by locks — APPROVED** [review]

**Comment by locks** [comment]
@rwjblue anything necessary to make merging and releasing easier?

**Pull request merged**